src/porepy/fracs/fracture.py

```diff
@@ -0,0 +1,369 @@
+            dist = pp.distances.pointset(self.pts)
+            dist /= np.amax(dist)
+            np.fill_diagonal(dist, np.inf)
+            mask = np.where(dist < enforce_pt_tol)[0]
+            mask = np.setdiff1d(mask, to_enforce, assume_unique=True)
+            self.remove_points(mask)
+
+        if check_convexity:
+            return self.is_convex() and self.is_planar(tol)
+        else:
+            return self.is_planar()
+
+    def remove_points(
+        self, ind: Union[np.ndarray, List[int]], keep_orig: bool = False
+    ) -> None:
+        """Remove points from the fracture definition.
+
+        Parameters
+        ----------
+            ind : numpy.ndarray , or similar
+                Indices of points to remove.
+            keep_orig : bool , optional
+                Defaults to False.
+                If True, keeps the original points in the attribute :attr:`~Fracture.orig_pts`.
+
+        """
+
+        self.pts = np.delete(self.pts, ind, axis=1)
+
+        if not keep_orig:
+            self.orig_pts = self.pts
+
+    def copy(self: FractureType) -> FractureType:
+        """Return a copy of the fracture with the current vertices.
+
+        Notes
+        -----
+            The original points (as given when the fracture was initialized)
+            will *not* be preserved.
+
+        Returns
+        -------
+            Fracture with the same points.
+        """
+        # deep copy points since mutable
+        p = np.copy(self.pts)
+        # TODO: Consider adding optional index parameter to this method.
+        return type(self)(p, index=self.index)
+
+    ###########################################################################################
+    ### Dimension-dependent, abstract methods
+    ###########################################################################################
+
+    @abc.abstractmethod
+    def sort_points(self) -> np.ndarray:
+        """Abstract method to sort the vertices as needed for geometric algorithms.
+
+        Returns
+        -------
+        numpy.ndarray(dtype=int)
+            The indices corresponding to the sorting.
+        """
+        pass
+
+    @abc.abstractmethod
+    def local_coordinates(self) -> np.ndarray:
+        """Abstract method for computing the vertex coordinates in a local system and its
+        local dimension `d`.
+        For now, `d` is expected to be `nd` - 1, i.e. the fracture to be of co-dimension 1.
+
+        Returns
+        -------
+        numpy.ndarray((nd-1) x npt)
+            The coordinates of the vertices in local dimensions.
+        """
+        pass
+
+    @abc.abstractmethod
+    def is_convex(self) -> bool:
+        """Abstract method for implementing a convexity check.
+
+        Returns
+        -------
+        bool
+            True if the fracture is convex, False otherwise.
+        """
+        pass
+
+    @abc.abstractmethod
+    def is_planar(self, tol: float = 1e-4) -> bool:
```

Comment:
In my understanding, "planar" implies 2d. Do we want to move this to PlanarFracture or generalize the method name?

Reply:
Maybe we can use `is_flat` instead?